opencti-platform/opencti-graphql/src/utils/stix-filtering/boolean-logic-engine.ts

```diff
@@ -0,0 +1,187 @@
+  if (adaptedFilterValues.length === 0) {
+    return false;
+  }
+  if (mode === 'and') {
+    // we need to find all of them or none of them
+    return (operator === 'eq' && adaptedFilterValues.every((v) => stixCandidates.includes(v)))
+      || (operator === 'not_eq' && adaptedFilterValues.every((v) => !stixCandidates.includes(v)))
+
+      // In real cases, there is only 1 filter value with the next operators (not much sense otherwise)
+      || (operator === 'lt' && adaptedFilterValues.every((v) => stixCandidates.some((c) => c < v)))
+      || (operator === 'lte' && adaptedFilterValues.every((v) => stixCandidates.some((c) => c <= v)))
+      || (operator === 'gt' && adaptedFilterValues.every((v) => stixCandidates.some((c) => c > v)))
+      || (operator === 'gte' && adaptedFilterValues.every((v) => stixCandidates.some((c) => c >= v)));
+  }
+
+  if (mode === 'or') {
+    // we need to find one of them or at least one is not found
+    return (operator === 'eq' && adaptedFilterValues.some((v) => stixCandidates.includes(v)))
+      || (operator === 'not_eq' && adaptedFilterValues.some((v) => !stixCandidates.includes(v)))
+
+      // In real cases, there is only 1 filter value with the next operators (not much sense otherwise)
+      || (operator === 'lt' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c < v)))
+      || (operator === 'lte' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c <= v)))
+      || (operator === 'gt' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c > v)))
+      || (operator === 'gte' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c >= v)));
+  }
+
+  return false;
+};
+
+/**
+ * Implementation of testGenericFilter for string values.
+ * String comparison is insensitive to case.
+ */
+export const testStringFilter = (filter: FilterExcerpt, stixCandidates: string[]) => {
+  const filterValuesLowerCase = filter.values.map((v) => v.toLowerCase());
+  const stixValuesLowerCase = stixCandidates.map((v) => v.toLowerCase());
+  return testGenericFilter<string>(filter, filterValuesLowerCase, stixValuesLowerCase);
+};
+
+/**
+ * Implementation of testGenericFilter for boolean values.
+ * Filter values are parsed as booleans
+ * The strings "true", "yes" or "1" are interpreted as true ; anything else is false.
+ */
+export const testBooleanFilter = (filter: FilterExcerpt, stixCandidate: boolean | null | undefined) => {
+  const filterValuesAsBooleans = filter.values.map((v) => v.toLowerCase() === 'true' || v.toLowerCase() === 'yes' || v === '1');
+  return testGenericFilter<boolean>(filter, filterValuesAsBooleans, toValidArray(stixCandidate));
+};
+
+/**
+ * Implementation of testGenericFilter for numerical values.
+ * Filter values are parsed as integers.
+ */
+export const testNumericFilter = (filter: FilterExcerpt, stixCandidate: number | null | undefined) => {
+  const filterValuesAsNumbers = filter.values.map((v) => parseInt(v, 10)).filter((n) => !Number.isNaN(n));
```

Comment:
By default the base is 10 for the parseInt. So it's not mandatory to add the second argument but we can let's it like that if you want to explicitly say the value.

Reply:
it's made mandatory by a linter rule https://eslint.org/docs/latest/rules/radix
Maybe an outdated rule, but still commonly used.